RecoHGCal/TICL/plugins/HGCDoublet.cc

```diff
@@ -23,9 +25,12 @@ bool HGCDoublet::checkCompatibilityAndTag(std::vector<HGCDoublet> &allDoublets,
       xi[j] = otherDoublet.innerX();
       yi[j] = otherDoublet.innerY();
       zi[j] = otherDoublet.innerZ();
+      seedi[j] = otherDoublet.seedIndex();
     }
     for (int j = 0; j < vs; ++j) {
-      ok[j] = areAligned(xi[j], yi[j], zi[j], xo, yo, zo, minCosTheta, minCosPointing, debug);
+      if (seedi[j] != seedIndex_)
+        continue;
+      ok[j] = areAligned(xi[j], yi[j], zi[j], xo, yo, zo, minCosTheta, minCosPointing, refDir, debug);
```

Comment:
apparently the "continue" above is leading to the static analyzer warning at https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d5a760/1333/llvm-analysis/report-687371.html#EndPath
This looks real

Reply:
yes thanks. bug fixed